sdk/storage/AzureStorageBlob/Source/Transport/URLSessionTransferManager.swift

```diff
@@ -85,15 +83,12 @@ internal final class URLSessionTransferManager: NSObject, TransferManager, URLSe
     private override init() {
         self.operationQueue = ResumableOperationQueue(name: "TransferQueue", delegate: nil, removeOnCompletion: true)
         self.transfers = [Transfer]()
-        self.logger = ClientLoggers.none
 
         super.init()
         operationQueue.delegate = self
         operationQueue.maxConcurrentOperationCount = 4
     }
 
-    // TODO: This will interfere with trying to use multiple BlobClients simultaneously. Find an alternate
-    // solution such that the minimal set (the NSPersistentContainer) is shared.
```

Comment:
Does this PR actually resolve this?

Reply:
I believe so, but I will test. Multiple blob clients will always use the single static TransferManager instance, with a single `NSPersistentContainer`. I believe this comment was originally referring to the plan of having multiple TransferManagers but letting them share the `NSPersistentContainer` so that we don't encounter CoreData concurrency problems. Now with a single TransferManager instance and a single `NSPersistentContainer` that should no longer be a problem.

Reply:
It could... depends on where and how things stateful things like the delegate get moved out of the Transfer Manager.

Reply:
> The question for me would be, is this a programming error? If so, perhaps an `assertionFailed` call so that way it crashes during debug development, but then mark the transfer failed and NOT crash during a production build.

I don't think I would call it that. As a naive developer, I might just do the following:

onApplicationStart:
- create Client A
- call resumeAll()

Then later in some other part of the application:
- create Client B
- call resumeAll()

Right now, resumeAll() resumes -all- transfers. Maybe we should make the client's *all methods only operate on transfers that match the client's restoration ID? That would make logical sense. But there would no longer be a handle to unconditionally pause/resume each and every transfer regardless of client. I'm going to put a little thought into this, because I can see use cases for both, and for the latter, I don't think it makes sense for those APIs to live on the client, since everything else the client does is scoped to that client.

Reply:
> The question for me would be, is this a programming error? If so, perhaps an `assertionFailed` call so that way it crashes during debug development, but then mark the transfer failed and NOT crash during a production build.

I added an assert for this condition as suggested. I think that's the most pragmatic approach.

Reply:
I'm just going to poke at the sample app and see if I can reproduce a coredata issue. If you have a set of reliable repro steps, please do share :)

Reply:
Pretty much, you start a bunch of parallel downloads and then jump to the upload page. It's almost like even if you try to always commit to the same context, if two threads try at the same time, rather than synchronize the operation, CoreData creates a new context and commits the change to that. With the parent and its child separated, CoreData validation fails when you try to save.